parser/parser.go

```diff
@@ -108,6 +128,60 @@ func Parse(r io.Reader, pkgName string) (*Report, error) {
 			// clear the current build package, so output lines won't be added to that build
 			capturedPackage = ""
 			seenSummary = false
+		} else if strings.HasPrefix(line, "Benchmark") {
+			// parse benchmarking info
```

Comment:
The parser is already pretty complicated, so I was thinking to keep this part as simple as possible. We just capture every benchmark that we find and add it to the list of benchmarks. We don't care about duplicates here yet.

Then in the formatter define a function `mergeBenchmarks`, which merges the duplicates and calculates the averages. The formatter can then iterate over this result when converting Report.Benchmarks to JUnitTestCase's.

This should allow you to refactor this `else if` branch down to just a couple of lines (creating a `Benchmark` from the `matches` and appending it to `benchmarks`).